test/test_utils/mbc_simulation_utils.jl

```diff
@@ -0,0 +1,574 @@
+    )
+
+# debugging code for inspecting objective functions -- ignore
+# TODO LK group by folders.
+const DOWNLOADS = joinpath(homedir(), "Downloads")
+function format_objective_function_file(filepath::String)
+    if !isfile(filepath)
+        println("Error: File '$filepath' does not exist.")
+        exit(1)
+    end
+
+    try
+        content = read(filepath, String)
+        content = replace(content, "+" => "+\n")
+        content = replace(content, "-" => "-\n")
+        write(filepath, content)
+    catch e
+        println("Error processing file '$filepath': $e")
+        exit(1)
+    end
+end
+
+function save_objective_function(model::DecisionModel, filepath::String)
+    open(filepath, "w") do file
+        println(file, "invariant_terms:")
+        println(file, model.internal.container.objective_function.invariant_terms)
+        println(file, "variant_terms:")
+        println(file, model.internal.container.objective_function.variant_terms)
+    end
+    format_objective_function_file(filepath)
+end
+
+function save_constraints(model::DecisionModel, filepath::String)
+    open(filepath, "w") do file
+        for (k, v) in model.internal.container.constraints
+            println(file, "Constraint Type: $(k)")
+            println(file, v)
+        end
+    end
+end
+# end debugging code
+
+function set_formulations!(template::ProblemTemplate,
+    sys::PSY.System,
+    device_to_formulation::Dict{Type{<:PSY.Device}, Type{<:PSI.AbstractDeviceFormulation}},
+)
+    for (device, formulation) in device_to_formulation
+        if !isempty(get_components(device, sys))
+            set_device_model!(template, device, formulation)
+        end
+    end
+    for (device, formulation) in DEFAULT_FORMULATIONS
+        if !haskey(device_to_formulation, device) && !isempty(get_components(device, sys))
+            set_device_model!(template, device, formulation)
+        end
+    end
+end
+
+# Layer of indirection to upgrade problem results to look like simulation results
+_maybe_upgrade_to_dict(input::AbstractDict) = input
+_maybe_upgrade_to_dict(input::DataFrame) =
+    SortedDict{DateTime, DataFrame}(first(input[!, :DateTime]) => input)
+
+read_variable_dict(
+    res::IS.Results,
+    var_name::Type{<:PSI.VariableType},
+    comp_type::Type{<:PSY.Component},
+) =
+    _maybe_upgrade_to_dict(read_variable(res, var_name, comp_type))
+read_parameter_dict(
+    res::IS.Results,
+    par_name::Type{<:PSI.ParameterType},
+    comp_type::Type{<:PSY.Component},
+) =
+    _maybe_upgrade_to_dict(read_parameter(res, par_name, comp_type))
+
+function _read_one_value(res, var_name, gentype, unit_name)
+    df = @chain begin
+        vcat(values(read_variable_dict(res, var_name, gentype))...)
+        @rsubset(:name == unit_name)
+        @combine(:value = sum(:value))
+    end
+    return df[1, 1]
+end
+
+function build_generic_mbc_model(sys::System;
+    multistart::Bool = false,
+    standard::Bool = false,
```

Comment:
Pretty cryptic, here and elsewhere. `model_thermal_standard`? Same to a lesser extent for `multistart`.